source/features/add-keyboard-shortcuts-to-comment-fields.js

```diff
@@ -49,10 +64,24 @@ export default function () {
 				event.preventDefault();
 			}
 		} else if (event.key === 'Escape') {
-			const cancelButton = select('.js-hide-inline-comment-form', field.form);
+			const inlineCancelButton = select('.js-hide-inline-comment-form', field.form);
+
+			// Cancel comment if inline, blur the field if it's a regular comment
+			if (field.value === '') {
+				blurAccessibly(field);
+			} else if (inlineCancelButton) {
+				inlineCancelButton.click();
```

Comment:
I lost so many comments because of this! I frequently write `:p` but github doesn't know this emoji and opens an autocomplete with `:package:` & co. So I type escape to hide the autocomplete and it closes the entire comment form with no warning and all my text is lost :(

Few suggestions:
- Check if the autocomplete is visible and do not trigger this action if it is
- If there's content, ask for confirmation before deleting all the content

Reply:
> If there's content, ask for confirmation before deleting all the content

[We do.](https://github.com/sindresorhus/refined-github/blob/dbb0f54a944e8f7e7946ddaf4223baf19c0a75f7/source/features/add-confirmation-to-comment-cancellation.js) Where does that happen exactly? Can you provide a URL where I can leave a comment that gets lost?

Are you on Chrome?

Do you have any disabled features

Reply:
I'm using Chrome.

I didn't disable any feature (well, until I found out about this issue and disabled add-keyboard-shortcuts-to-comment-fields which stops this behavior from happening).

I was able to repro pressing escape on this very comment. I don't see any warning saying that i'm about to discard the comment when pressing escape or pressing cancel.

Reply:
Repro steps:
 - Reply to this comment
 - Press escape
 - The whole reply box disappears with no confirmation

[Before pressing escape]

![image](https://user-images.githubusercontent.com/197597/37169233-325672f0-22bc-11e8-911d-440e94fe1bb5.png)

[After pressing escape]

<img width="720" alt="screen shot 2018-03-08 at 10 33 57 am" src="https://user-images.githubusercontent.com/197597/37169249-3ccf641c-22bc-11e8-9cd8-19bf3e1bd510.png">

Reply:
As I'm playing around with it, it doesn't always repro. Sometimes I do get a confirmation dialog, sometimes I don't. I can't see any error in the js console.

Reply:
I can't repro at all, I always get the confirmation

![gif](https://user-images.githubusercontent.com/1402241/37169786-82b8efd2-233b-11e8-8db0-353fd420ee2a.gif)

Reply:
Okay this is crazy, I can no longer repro... I'll reply here if I can get a repro and figure out what could happen. Sorry for the false alarm.